drv/transceivers-server/src/main.rs

```diff
@@ -160,16 +190,117 @@ impl idl::InOrderTransceiversImpl for ServerImpl {
+            // TODO: remove conditional compilation path once sidecar-a is sunset
+            cfg_if::cfg_if! {
+                if #[cfg(target_board = "sidecar-a")] {
+                    let _ = self.leds.update_led_state(presence);
+                } else {
+                    self.leds.update_led_state(presence).unwrap();
+                }
+            }
+
+            self.modules_present = presence;
+            ringbuf_entry!(Trace::ModulePresenceUpdate(presence));
+        }
+
+        let next_deadline = sys_get_timer().now + TIMER_INTERVAL;
+        sys_set_timer(Some(next_deadline), TIMER_NOTIFICATION_MASK)
+    }
+}
+
 #[export_name = "main"]
 fn main() -> ! {
     loop {
-        let mut buffer = [0; idl::INCOMING_SIZE];
+        // This is a temporary workaround that makes sure the FPGAs are up
+        // before we start doing things with them. A more sophisticated
+        // notification system will be put in place.
+        let seq = Sequencer::from(SEQ.get_task_id());
+        loop {
+            let ready = seq.front_io_phy_ready();
+            match ready {
+                Ok(true) => {
+                    ringbuf_entry!(Trace::FrontIOReady(true));
+                    break;
+                }
+                Err(SeqError::NoFrontIOBoard) => {
+                    ringbuf_entry!(Trace::FrontIOSeqErr(
+                        SeqError::NoFrontIOBoard
+                    ));
+                    break;
+                }
+                _ => {
+                    ringbuf_entry!(Trace::FrontIOReady(false));
+                    userlib::hl::sleep_for(10)
+                }
+            }
+        }
+
         let transceivers = Transceivers::new(FRONT_IO.get_task_id());
+        let leds = Leds::new(
+            &i2c_config::devices::pca9956b_front_leds_left(I2C.get_task_id()),
+            &i2c_config::devices::pca9956b_front_leds_right(I2C.get_task_id()),
+        );
 
-        let mut server = ServerImpl { transceivers };
+        let mut server = ServerImpl {
+            transceivers,
+            leds,
+            modules_present: 0,
+            led_error: Default::default(),
+        };
 
+        ringbuf_entry!(Trace::LEDInit);
+
+        server.transceivers.enable_led_controllers().unwrap();
+
+        // Errors are being suppressed here due to a miswiring of the I2C bus at
+        // the LED controller parts. They will not be accessible without rework
+        // to older hardware, and newer (correct) hardware will be replacing the
+        // hold stuff very soon.
+        // TODO: remove conditional compilation path once sidecar-a is sunset
+        cfg_if::cfg_if! {
+            if #[cfg(target_board = "sidecar-a")] {
+                let _ = server.leds.initialize_current();
+                let _ = server.leds.turn_on_system_led();
+            } else {
+                server.leds.initialize_current().unwrap();
+                server.leds.turn_on_system_led().unwrap();
```

Comment:
(For context, I'm fine panicking here, because if startup fails, there's not much else we can do...)

Reply:
Thinking on this more, I don't think we should panic here. I would just as soon hold a piece of state indicating that LEDs were not initialized (and therefore we should not try to update them) and then proceed as normal. We could then build out a recovery strategy if we so choose, but in reality if we fail to initialize the LEDs but can still talk to the FPGAs, we likely have a problem limited in scope to the PCA9956Bs. Following my logic above, I don't think LED failure should not impede transceiver activities.